Mirror magni_robot in loki_robot

rohbotics opened the pull request:
Fixes #1 

This should significantly simplify the whole loki setup situation from the mess it is right now.
It still isn't ideal because of all the code duplication, but we can work on fixing that.

Please look for any errors, as I probably screwed up somewhere, and I haven't tested this on anything.

rohbotics committed:
remove old ur launch stuff, start moving to metapackage

This is the first step in resolving issue #1

rohbotics committed:
add teleop config to loki_teleop

rohbotics committed:
added loki_description for urdf

rohbotics committed:
depend on keyboard teleop to make sure it gets installed

rohbotics committed:
install meshes as well

rohbotics committed:
added launch files for base bringup

rohbotics committed:
add loki_nav, fix cmake errors

rohbotics committed:
mirror relevant parts of magni_demos to loki_demos

jim-v commented on loki_description/launch/description.launch:
```diff
@@ -0,0 +1,6 @@
+<launch>
+  <arg name="gui" default="False" />
+  <param name="use_gui" value="$(arg gui)"/>
+  <param name="robot_description" textfile="$(find magni_description)/urdf/loki.urdf" />
```
Seems to reference magni_description directory

rohbotics replied:
Fixed, also greped for magni and removed all references to it.

rohbotics committed:
remove magni refs in loki_description

jim-v reviewed (approved):
Looks OK to me, but I don't have the hardware to test on.

rohbotics committed:
make metapackage pull in all the subpackages

you need to use exec_depend on all the subpackages in the metapackage
otherwise the installing the metapackage won't pull in the others

rohbotics commented:
Tested on real hardware: teleop works, sonars come up, and urdf renders in rviz.

![img_20171227_130820875](https://user-images.githubusercontent.com/2825327/34393450-57beb760-eb07-11e7-8839-33cf117a932b.jpg)
![screenshot from 2017-12-27 13-08-52](https://user-images.githubusercontent.com/2825327/34393455-62c08a6c-eb07-11e7-9fb5-ae1e813ce447.png)

rohbotics committed:
fix urdf mesh refrences
